src/main/java/pl/touk/sputnik/connector/ConnectorFacadeFactory.java

```diff
@@ -1,27 +1,25 @@
 package pl.touk.sputnik.connector;
 
 import org.jetbrains.annotations.NotNull;
+
 import pl.touk.sputnik.connector.gerrit.GerritFacadeBuilder;
 import pl.touk.sputnik.connector.stash.StashFacadeBuilder;
 
 public class ConnectorFacadeFactory {
     public static final ConnectorFacadeFactory INSTANCE = new ConnectorFacadeFactory();
 
-    private static final String GERRIT = "gerrit";
-    private static final String STASH = "stash";
-
     GerritFacadeBuilder gerritFacadeBuilder = new GerritFacadeBuilder();
     StashFacadeBuilder stashFacadeBuilder = new StashFacadeBuilder();
 
     @NotNull
-    public ConnectorFacade build(String key) {
-        switch (key) {
+    public ConnectorFacade build(ConnectorType type) {
```

Comment:
OK, I also changed the type of the exception thrown by this method. Previously IllegalArgumentException crashed Sputnik, now GeneralOptionNotSupportedException is supported by Main.getConnectorFacade()